src/USER-SICRACK/fix_sicrack.cpp

```diff
@@ -0,0 +1,318 @@
+
+  xtip = atof(arg[3]);  // Give coords in (rotated) lattice units, not in Ångstrøm
+  ytip = atof(arg[4]);
+  mu = atof(arg[5]);   // Shear modulus, in GPa
+  vu = atof(arg[6]);  // Poisson ratio
+
+  int iarg = 7;
+
+  while (iarg < narg) {
+    if (strcmp(arg[iarg],"M1") == 0) {
+      if (iarg+3 > narg) error->all(FLERR,"Illegal fix sicrack command");
+      else {
+	K1 = atof(arg[iarg+1]);  // Mode I stress-intensity factor, in MPa*sqrt(m)
+	dK1 = atof(arg[iarg+2]);  // Increase in Mode I stress-intensity factor, in MPa*sqrt(m)
+	iarg += 3;
+      }
+    } else if (strcmp(arg[iarg],"M2") == 0) {
+      if (iarg+3 > narg) error->all(FLERR,"Illegal fix sicrack command");
+      else {
+	K2 = atof(arg[iarg+1]);  // Mode II stress-intensity factor, in MPa*sqrt(m)
+	dK2 = atof(arg[iarg+2]);  // Increase in Mode II stress-intensity factor, in MPa*sqrt(m)
+	iarg += 3;
+      }
+    } else if (strcmp(arg[iarg],"M3") == 0) {
+      if (iarg+3 > narg) error->all(FLERR,"Illegal fix sicrack command");
+      else {
+	K3 = atof(arg[iarg+1]);  // Mode III stress-intensity factor, in MPa*sqrt(m)
+	dK3 = atof(arg[iarg+2]);  // Increase in Mode III stress-intensity factor, in MPa*sqrt(m)
+	iarg += 3;
+      }
+    } else if (strcmp(arg[iarg],"T") == 0) {
+      if (iarg+3 > narg) error->all(FLERR,"Illegal fix sicrack command");
+      else {
+	T = atof(arg[iarg+1]);  // T-stress, in MPa
+	dT = atof(arg[iarg+2]);  // Increase in T-stress, in MPa
+	iarg += 3;
+      }
+    } else error->all(FLERR,"Illegal fix sicrack command");
+  }
+
+  PI = 4.0*atan(1.0);
+  Xk = 3.0-4.0*vu;
+
+  double xlattice = domain->lattice->xlattice;  // Get lattice scale factors
+  double ylattice = domain->lattice->ylattice;
+  xtip *= xlattice;  // Set the true coords of crack front
+  ytip *= ylattice;
+  //  printf("K1: %f, dK1: %f, K2: %f, dK2: %f, K3: %f, dK3: %f, T: %f, dT: %f\n", K1, dK1, K2, dK2, K3, dK3, T, dT);
+
+  // perform initial allocation of atom-based array
+  // register with Atom class
+
+  xoriginal = NULL;
+  grow_arrays(atom->nmax);
+  atom->add_callback(0);
+  atom->add_callback(1);
+
+  // xoriginal = initial unwrapped positions of atoms
+
+  double **x = atom->x;
+  int *image = atom->image;
+  int *mask = atom->mask;
+  int nlocal = atom->nlocal;
+
+  for (int i = 0; i < nlocal; i++) {
+    if (mask[i] & groupbit) domain->unmap(x[i],image[i],xoriginal[i]);
+    else xoriginal[i][0] = xoriginal[i][1] = xoriginal[i][2] = 0.0;
+  }
+
+}
+
+/* ---------------------------------------------------------------------- */
+
+FixSICrack::~FixSICrack()
+{
+  // unregister callbacks to this fix from Atom class
+ 
+  atom->delete_callback(id,0);
+  atom->delete_callback(id,1);
+
+  // delete locally stored arrays
+
+  memory->destroy(xoriginal);
+}
+
+/* ---------------------------------------------------------------------- */
+
+
+int FixSICrack::setmask()
+{
+  int mask = 0;
+  mask |= LAMMPS_NS::FixConst::INITIAL_INTEGRATE;

```

Comment:
no need to use the `LAMMPS_NS::` prefix here since you already have a `using LAMMPS_NS;`